src/Corax/Queries/BinaryMatch.Boolean.cs

```diff
@@ -203,7 +203,7 @@ static int FillFunc(ref BinaryMatch<TInner, TOuter> match, Span<long> matches)
 
                 // Given that we are going to be calling Fill, which would try to fill the buffer as much as it can there is no
                 // reason why we are not going to try to fill it here but avoid going overboard for leftovers.
-                while (leftoverMatches.Length > 0)
+                while (leftoverMatches.Length > 16)
```

Comment:
Can you please elaborate on this change? It's interesting that the condition has changed from `> 0` to `> 16`.

Reply:
Suppose the following query: 

```
SubCond(1) AND SubCond(2)
```

Imagine the following scenario:

- You have 1 element left and request more elements from the right branch of an `And` operation. 
- Now you execute a fill on `SubCond(2)` and get returned. 
- The results are in `SubCond(1)` , therefore you gonna get 1 more element... 

Essentially the worst case scenario is as many fills as elements in `SubCond(2)`. This condition what does is definining a minimum batch of work, if there is less than 16 elements to fill, it is not worth it to continue trying because the fill call would be higher than the probability of filling that block. Probably that number will increase in the future, so far 16 is about right.

Reply:
Can we add this explanation as a comment?

Reply:
Done